lib/ruby_lsp/parameter_scope.rb

```diff
@@ -0,0 +1,33 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyLsp
+  class ParameterScope
+    extend T::Sig
+
+    sig { returns(T.nilable(ParameterScope)) }
+    attr_reader :parent
+
+    sig { params(parent: T.nilable(ParameterScope)).void }
+    def initialize(parent = nil)
```

Comment:
We don't pass an argument when we instantiate it for the first time in `SemanticHighlighting#initialize`. It would also not be correct to always require a parent given that the global scope doesn't have one.